frontend/assignments/3/src/stocklist/StockList.tsx

```
@@ -0,0 +1,103 @@
+
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { RootState, AppDispatch } from '../Slices/store';
+import { setStocks, addToWishlist } from '../Slices/stockSlice';
+import Pagination from '@mui/material/Pagination';
+import Stack from '@mui/material/Stack';
+import "./StockList.scss";
+import { Link } from 'react-router-dom';
+
+
+interface Stock {
+ id: string;
+ name: string;
+ price: number;
+}
+
+const StocksList: React.FC = () => {
+ const dispatch = useDispatch<AppDispatch>();
+ const allStocks = useSelector((state: RootState) => state.stocks.stocks);
+ const [currentPage, setCurrentPage] = useState(1);
+ const itemsPerPage = 10;
+
+ useEffect(() => {
+    const fetchStocks = async () => {
+      try {
+        const response = await fetch('https://dev-6hy1cukrs46j7f1.api.raw-labs.com/mock/json-api');
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
```

lets throw an error response to the user.